ls1intum/Artemis
# `Communication`: Migrate exercise/lecture posts to messages

### Checklist
#### General



### Motivation and Context


With #6548, exercises and lectures support channel messages as well of course posts to be displayed on the exercise/lecture pages if communication or messaging is enabled. 

### Description

This PR introduces a migration script, that moves lecture and exercise posts from courses with both communication and messaging enabled to the new system using channels. It creates the needed lecture/exercise channels and adds all course members as participants. Exercise/Lecture posts are moved to the belonging channels.

### Steps for Testing

Prerequisites:
- 1 Instructor
- A course with exercise/lecture posts and communication/messaging enabled. (If there is co course available, create or choose a course with only communication enabled, create a few example lecture/exercise posts and then enable messaging for the course).

1. Roll out this branch
2. For the test course, within the messages tab, there should be new channels for exercises/lectures
3. Channel messages should be visible on exercise/lecture pages
4. Courses, that do not have both communication and messaging enabled should still look as before.


### Review Progress






#### Code Review
#### Manual Tests
